src/state/queries/post-thread.ts

```diff
@@ -0,0 +1,200 @@
+import {AppBskyFeedDefs, AppBskyFeedGetPostThread} from '@atproto/api'
+import {useQuery, QueryClient, useQueryClient} from '@tanstack/react-query'
+import {useSession} from '../session'
+import {RQKEY as POST_RQKEY, getCachedPost} from './post'
+import {ThreadViewPreference} from '../models/ui/preferences'
+
+export const RQKEY = (uri: string) => ['post-thread', uri]
+type ThreadViewNode = AppBskyFeedGetPostThread.OutputSchema['thread']
+
+export interface PostThreadSkeletonCtx {
+  depth: number
+  isHighlightedPost?: boolean
+  hasMore?: boolean
+  showChildReplyLine?: boolean
+  showParentReplyLine?: boolean
+}
+
+export type PostThreadSkeletonPost = {
+  type: 'post'
+  _reactKey: string
+  uri: string
+  parent?: PostThreadSkeletonNode
+  replies?: PostThreadSkeletonNode[]
+  viewer?: AppBskyFeedDefs.ViewerThreadState
+  ctx: PostThreadSkeletonCtx
+}
+
+export type PostThreadSkeletonNotFound = {
+  type: 'not-found'
+  _reactKey: string
+  uri: string
+  ctx: PostThreadSkeletonCtx
+}
+
+export type PostThreadSkeletonBlocked = {
+  type: 'blocked'
+  _reactKey: string
+  uri: string
+  ctx: PostThreadSkeletonCtx
+}
+
+export type PostThreadSkeletonUnknown = {
+  type: 'unknown'
+  uri: string
+}
+
+export type PostThreadSkeletonNode =
+  | PostThreadSkeletonPost
+  | PostThreadSkeletonNotFound
+  | PostThreadSkeletonBlocked
+  | PostThreadSkeletonUnknown
+
+export function usePostThreadQuery(uri: string | undefined) {
+  const {agent} = useSession()
+  const queryClient = useQueryClient()
+  return useQuery<PostThreadSkeletonNode, Error>(
+    RQKEY(uri || ''),
+    async () => {
+      const res = await agent.getPostThread({uri: uri!})
+      if (res.success) {
+        hydrateCache(queryClient, res.data.thread)
+        return threadViewToSkeleton(res.data.thread)
+      }
+      return {type: 'unknown', uri: uri!}
+    },
+    {enabled: !!uri},
+  )
+}
+
+export function sortThreadSkeleton(
```

Comment:
These functions are so clear 🥺 I love them

Reply:
oh yeah why dont you marry them

Reply:
Would love a comment overview of how this function sorts the posts for future contributors:
```
// 1. If node is not a post return it unmodified, otherwise continue.
// 2. If the node has replies, start the sorting process.
// 3. Retrieve the cached post for the current node using its URI.
// 4. Sort the replies array in place using a custom comparator.
// 5. Inside the comparator, give priority to posts over non-posts.
// 6. Retrieve cached posts for both comparison nodes using their URIs.
// 7. If the cached post for 'a' is not found, it is moved down in the sort order.
// 8. If the cached post for 'b' is not found, it is moved up in the sort order.
// 9. Check if the author of post 'a' is the original poster (OP) and the same for post 'b'.
// 10. If both are by the OP, sort by the timestamp 'indexedAt' to get the oldest post.
// 11. If only 'a' is by the OP, it gets higher priority.
// 12. If only 'b' is by the OP, 'a' gets higher priority.
// 13. If user preference is to prioritize followed users, compare the following status of each post's author.
// 14. Sort by followed status, giving priority to followed user's posts.
// 15. If sorting by 'oldest', sort by 'indexedAt' to get the oldest post.
// 16. If sorting by 'newest', sort by 'indexedAt' in reverse to get the newest post.
// 17. If sorting by 'most-likes', compare like counts; in case of a tie, use 'indexedAt' to prioritize newer posts.
// 18. If sorting by 'random', return a random order by using Math.random.
// 19. Default to sorting by 'indexedAt' in reverse if no other conditions are met.
// 20. Recursively apply the sorting function to each reply.
// 21. Finally, return the sorted node.
```

Reply:
I'm not a huge fan of describing code a second time in the comments -- only summarizing or explaining